Add auto-correct for Style/RaiseArgs cop

Drenmi opened the pull request:
This change adds auto-correct for `Style/RaiseArgs`. I also took the opportunity to add some more documentation to the cop.

Examples:

```
raise StandardError.new("message")
```

will be corrected to:

```
raise StandardError, "message"
```

... and ...

```
raise StandardError, "message"
raise StanrardError, arg1, arg2, arg3
```

will be corrected to:

```
raise StandardError.new("message")
raise StandardError.new(arg1, arg2, arg3)
```

bbatsov commented on lib/rubocop/cop/style/raise_args.rb:
```diff
@@ -21,6 +39,33 @@ def on_send(node)
+          ->(corrector) { corrector.replace(node.source_range, replacement) }
+        end
+
+        def correction_compact_to_exploded(node)
```
Why not `correct_compact_to_exploded`?

Drenmi replied:
Actually, I started with that, but decided to remove the imperative verb, since the method returns the corrected source, and doesn't actually do the correction itself (after refactoring to reduce duplication.)

Other options could be `#corrected_compact_to_exploded` or `#replacement_compact_to_exploded`?

bbatsov replied:
Got it. You're reasoning makes sense.

bbatsov commented on lib/rubocop/cop/style/raise_args.rb:
```diff
@@ -4,7 +4,25 @@
+      #
+      # @example
+      #
+      #   @bad
```
@bad => #bad

bbatsov commented on lib/rubocop/cop/style/raise_args.rb:
```diff
@@ -4,7 +4,25 @@
+      #   @good
+      #   raise StandardError, message
+      #   fail message
+      #   raise RuntimeError.new(arg1, arg2, arg3)
```
Maybe you should add 2 sets of examples for the two config styles?

bbatsov commented:
The changes look OK overall. You'll have to rebase on top of the current `master`.

Drenmi commented:
@bbatsov: Made corrections according to your comment, rebased, and extracted the message strings into frozen constants for consistency with the rest of the code base.

bbatsov commented:
Looks good. Just squash the two commits together and we're good to go.

Drenmi commented:
@bbatsov: *squashed*
